pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -903,6 +918,26 @@ func getFailingSyncSets(syncStatuses []hiveintv1alpha1.SyncStatus) []string {
 	return failures
 }
 
+func (r *ReconcileClusterSync) setFirstSuccessTime(syncStatuses []hiveintv1alpha1.SyncStatus, cd *hivev1.ClusterDeployment, clusterSync *hiveintv1alpha1.ClusterSync, logger log.FieldLogger) {
+	if reflect.ValueOf(cd.Status.InstalledTimestamp).IsNil() {
```

Comment:
Why do we need to use reflection here?
```suggestion
	if cd.Status.InstalledTimestamp == nil {
```